internal/pkg/template/templates/addons/ddb/env/access_policy.yml

```
@@ -0,0 +1,54 @@
+Parameters:
+  App:
+    Type: String
+    Description: Your application's name.
+  Env:
+    Type: String
+    Description: The environment name your service, job, or workflow is being deployed to.
+  Name:
+    Type: String
+    Description: Your workload's name.
+Resources:
+  {{logicalIDSafe .Name}}AccessPolicy:
+    Metadata:
+      'aws:copilot:description': 'An IAM ManagedPolicy for your service to access the {{.Name}} db'
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: !Sub
+        - Grants CRUD access to the Dynamo DB table ${Table}
+        - Table: { Fn::ImportValue: { Fn::Sub: "${App}-${Env}-{{logicalIDSafe .Name}}TableName" } }
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: DDBActions
+            Effect: Allow
+            Action:
+              - dynamodb:BatchGet*
+              - dynamodb:DescribeStream
+              - dynamodb:DescribeTable
+              - dynamodb:Get*
+              - dynamodb:Query
+              - dynamodb:Scan
+              - dynamodb:BatchWrite*
+              - dynamodb:Create*
+              - dynamodb:Delete*
+              - dynamodb:Update*
+              - dynamodb:PutItem
+            Resource: 
+              Fn::ImportValue: !Sub "${App}-${Env}-{{logicalIDSafe .Name}}ARN"
```

Should this be something like:
```suggestion
              Fn::ImportValue: !Sub "${App}-${Env}-{{logicalIDSafe .Name}}TableArn"
```
Otherwise it would render something like `demo-test-frontendARN` which doesn't seem specific enough 💭

---

Just to make sure I'm on the same page, there will be:
1. An environment addon that has a `DynamoDB::Table` resource
2. But the resource name/logical ID/exports won't be something like `{App}{Env}Table` instead it will still be specific to a particular service? such as `{App}{Env}{Workload}Table`

---

> An environment addon that has a DynamoDB::Table resource

This is correct

> But the resource name/logical ID/exports won't be something like {App}{Env}Table instead it will still be specific to a particular service? such as {App}{Env}{Workload}Table

No it will be `{App}-{Env}-{DDB-Addon-Name}-TableArn`. The data in `{{ .Name }}` is the storage name, that is the value of `--name` in `copilot storage init --name`.

This `access_policy.yml` file is added to `copilot/<workload>/addons/access_policy.yml` at the same time as `copilot/environments/addons/ddb.yml` when the user create an env-level ddb.